components/layout/display_list_builder.rs

```diff
@@ -1834,6 +1841,8 @@ impl FragmentDisplayListBuilding for Fragment {
         let stacking_relative_content_box =
             self.stacking_relative_content_box(stacking_relative_border_box);
 
+        let clipping_region = clip_for_border_radius(&self.style, &stacking_relative_content_box);
```

Comment:
Why `stacking_relative_border_box`? I'd have expected this to use `clip`.

Also, what about the other uses for the `clip` rect? Should they account for border-radius too?